### DISCUSSION TITLE
Using CD `hook-delete-policy` on Workflow resource

### DISCUSSION THREAD

**Discussion by masterginger** [argoproj/argo-workflows#12280] [Q&A] [answered]
Hi all,

We were looking at [hook deletion policy](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_hooks/#hook-deletion-policies), (which mentioned it could be applied to `Workflow` resources) and tried to add the following annotations to the `Workflow` resource:
```yaml
argocd.argoproj.io/hook: PostSync
argocd.argoproj.io/hook-delete-policy: HookSucceeded
```

However, we observed that after the `Workflow` resource was created and synced, it was deleted **before** the workflow itself actually completed. There were still pods (for workflow steps) running. As a result, the workflow execution was interrupted (the pods for workflow steps were deleted along with the workflow resource itself) and most of the workflow steps were never run.

Is this the expected behavior?  or adding hook delete policy to `Workflow` resource is not a valid use case in the first place?

Any feedback is appreciated!

**Comment by Joibel** [comment, marked as answer]
This is a question for [ArgoCD](https://github.com/argoproj/argo-cd), not argo workflows. Workflows is not implementing this feature, and argocd is the tool that is deleting the workflow prematurely. Please raise your question there.

**Reply by masterginger** [reply]
Moved to https://github.com/argoproj/argo-cd/discussions/16468
